modules/shortcodes/slideshow.php

```diff
@@ -216,52 +187,55 @@ function shortcode_callback( $attr ) {
 	 *
 	 * Returns the necessary markup and js to fire a slideshow.
 	 *
-	 * @param $attr array Attributes for the slideshow.
+	 * @param array $attr Attributes for the slideshow.
 	 *
 	 * @uses $this->enqueue_scripts()
 	 *
 	 * @return string HTML output.
 	 */
-	function slideshow_js( $attr ) {
-		// Enqueue scripts
+	public function slideshow_js( $attr ) {
+		// Enqueue scripts.
 		$this->enqueue_scripts();
 
 		$output = '';
 
 		if ( defined( 'JSON_HEX_AMP' ) ) {
-			// This is nice to have, but not strictly necessary since we use _wp_specialchars() below
-			$gallery = json_encode( $attr['gallery'], JSON_HEX_TAG | JSON_HEX_AMP | JSON_HEX_APOS | JSON_HEX_QUOT ); // phpcs:ignore PHPCompatibility
+			// This is nice to have, but not strictly necessary since we use _wp_specialchars() below.
+			$gallery = wp_json_encode( $attr['gallery'], JSON_HEX_TAG | JSON_HEX_AMP | JSON_HEX_APOS | JSON_HEX_QUOT ); // phpcs:ignore PHPCompatibility
 		} else {
-			$gallery = json_encode( $attr['gallery'] );
+			$gallery = wp_json_encode( $attr['gallery'] );
 		}
 
 		$output .= '<p class="jetpack-slideshow-noscript robots-nocontent">' . esc_html__( 'This slideshow requires JavaScript.', 'jetpack' ) . '</p>';
+
+		/*
+		 * The input to json_encode() above can contain '&quot;'.
+		 *
+		 * For calls to json_encode() lacking the JSON_HEX_AMP option,
+		 * that '&quot;' is left unaltered.  Running '&quot;' through esc_attr()
+		 * also leaves it unaltered since esc_attr() does not double-encode.
+		 *
+		 * This means we end up with an attribute like
+		 * `data-gallery="{&quot;foo&quot;:&quot;&quot;&quot;}"`,
+		 * which is interpreted by the browser as `{"foo":"""}`,
+		 * which cannot be JSON decoded.
+		 *
+		 * The preferred workaround is to include the JSON_HEX_AMP (and friends)
+		 * options, but these are not available until 5.3.0.
```

Comment:
Until PHP 5.3? Could you note that and a `@todo Update when minimum is PHP5.3+` so we can come back and clean it up once WP 5.2 is our minimum.

(I realize this was originally present, but might as well clarify while we're touching it).

Reply:
👍Done in 70c475105f972302ad33579f8d62ca04c3df9ec3